pilot/proxy/envoy/discovery.go

```diff
@@ -27,12 +27,78 @@ import (
 
 	restful "github.com/emicklei/go-restful"
 	"github.com/golang/glog"
-	multierror "github.com/hashicorp/go-multierror"
-
+	"github.com/hashicorp/go-multierror"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"istio.io/istio/pilot/model"
 	"istio.io/istio/pilot/proxy"
 )
 
+const (
+	metricsPath          = "/metrics"
+	metricsNamespace     = "pilot"
+	metricsSubsystem     = "discovery"
+	metricLabelCacheName = "cache_name"
+	metricLabelMethod    = "method"
+)
+
+var (
+	cacheSizeGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metricsNamespace,
+			Subsystem: metricsSubsystem,
+			Name:      "cache_size",
+			Help:      "Current size of a single cache within Pilot",
```

Comment:
can you mention units? or is that hard to compute? (kb/mb/etc.)

Reply:
Are you suggesting to just indicate the units in the `Help` comment? done.